### PR TITLE
CSE-861 show shopware attribute labels

### PR DESCRIPTION
**Pull request by voodoods**
**Jira Issue**

https://8select.atlassian.net/browse/CSE-861

**Summary**
- show values beside labels in the dropdown selection of attribute mapping

**Checklist**

- [x] PR title is prefixed with Jira Issue Id - e.g. CSE-42
- [x] Add feature / bug label
- [x] Unit tests for new / changed logic exist OR no logic changes are passing

bildschirmfoto 2018-06-27 um 12 03 14

### CODE REVIEW

**Comment by voodoods** [comment]
Note: I wasn't able to display the proper label in the column after the selection via combobox. It turns out that the records (which contain the labels we want to display) are loaded asynchronously to the column. However, proper async methods that could be used in a custom renderer function such are only added in ExtJS v6.0+, where Shopware uses ExtJS 4.1.x, see also: https://www.sencha.com/blog/asynchronous-javascript-promises/

**Comment by akleiber** [comment]
I don't understand why this should not be possible.
You could just extend what is returned from the API call and add an `shopwareAttributeLabel` like the `eightselectAttributeLabel`

bildschirmfoto 2018-06-18 um 09 42 51

**Comment by voodoods** [comment]
@akleiber I thought so too but that's not how ExtJS works.

**TL;DR** 
After more research and tweaking around with the stores a lot I finally got a working solution 👍  Take a look at the last commit again to find out more. Screenshot in the summary has been updated.

**Long version:** 
1. The API call you talk about accesses the `EightselectAttribute` store, which does not contain the labels for `shopwareAttribute` as you can see in your screenshot - it's just the column names, which by themselves are not of much use to us; What we need are the labels to those column names
2. Our `grid` component uses this `EightselectAttribute` store as main data source; the labels that match these column names are part of a different store (the `ShopwareAttribute` store) 
3. We can load the `ShopwareAttribute` but we have to do it asynchronously via `store.load()` 
3. However, our `grid` component will not wait for asynchronous data to be loaded - it will render immediately as soon as its own data is ready
 
The solution was to pre-load the data from the external store object before our `grid` renders (see `beforerender()` method in the changes). Now, after our component has been refreshed with the preloaded data from our `ShopwareAttribute`, we can access its contents.

There may be a more elegant solution to this, however I wasn't able to find it :-/

**Comment by voodoods** [comment]
@Wytrykus fixed it. The error was caused by the field `s_articles_details.ean` for which there is apparently no `label` found on initialization of the mapping window. I now simply return the fieldname if no label can be found, so it won't break the modal anymore.

Will look into the issue of the missing label but to not waiste any more time, you can recheck it in the meantime 👍
